In [39]: one.author
Out[39]: <Author: Mike Q mike@cd.com>

In [40]: two = Book.objects.get(id=2)

In [41]: two
Out[41]: <Book: Java hi>

In [42]: two.author
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-42-4b65de81f126> in <module>()
----> 1 two.author

AttributeError: 'Book' object has no attribute 'author'

In [43]: two.author = Author.objects.get(id=2)

In [44]: two.author
Out[44]: <Author: Speros  >

In [45]: first3 = Book.objects.all()[:3]

In [46]: first3
Out[46]: <QuerySet [<Book: C Sharp hi>, <Book: Java hi>, <Book: Python hi>]>

In [47]: first3.author = Author.objects.get(id=2)

In [48]: first3.author = Author.objects.get(id=3)

In [49]: two.author
Out[49]: <Author: Speros  >

In [50]: first3.author = Author.objects.get(id=3)

In [51]: first3.save()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-51-e43198fd90ed> in <module>()
----> 1 first3.save()

AttributeError: 'QuerySet' object has no attribute 'save'

In [52]: books = Book.objects.all()

In [53]: books
Out[53]: <QuerySet [<Book: C Sharp hi>, <Book: Java hi>, <Book: Python hi>, <Book: PHP hi>, <Book: Ruby hi>]>

In [54]: for i in books[0:4]:
    ...:     i.author = Author.objects.get(id=3)
    ...:     

In [55]: two.author
Out[55]: <Author: Speros  >

In [56]: two = Book.objects.get(id=2)

In [57]: two
Out[57]: <Book: Java hi>

In [58]: two.author
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-58-4b65de81f126> in <module>()
----> 1 two.author

AttributeError: 'Book' object has no attribute 'author'

In [59]: for i in books[0:4]:
    ...:     i.author = Author.objects.get(id=3)
    ...:     

In [60]: 

In [60]: two.author
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-60-4b65de81f126> in <module>()
----> 1 two.author

AttributeError: 'Book' object has no attribute 'author'

In [61]: Book.objects.get(id=1).author
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-61-286c6a99014a> in <module>()
----> 1 Book.objects.get(id=1).author

AttributeError: 'Book' object has no attribute 'author'

In [62]: one = Book.objects.get(id=1)

In [63]: speros = Author.objects.get(id=2)

In [64]: speros
Out[64]: <Author: Speros  >

In [65]: one.author = speros

In [66]: one.author
Out[66]: <Author: Speros  >

In [67]: Book.objects.get(id=1).author
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-67-286c6a99014a> in <module>()
----> 1 Book.objects.get(id=1).author

AttributeError: 'Book' object has no attribute 'author'

In [68]: Jay = Author.objects.get(id=5)

In [69]: Jay
Out[69]: <Author: Jay  >

In [70]: Books = Book.objects.all()

In [71]: Books
Out[71]: <QuerySet [<Book: C Sharp hi>, <Book: Java hi>, <Book: Python hi>, <Book: PHP hi>, <Book: Ruby hi>]>

In [72]: for i in Books[0:4]:
    ...:     i.author = Jay
    ...:     

In [73]: book3 = Book.objects.get(id=3)

In [74]: book3
Out[74]: <Book: Python hi>

In [75]: book3.author
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-75-587a05487b6c> in <module>()
----> 1 book3.author

AttributeError: 'Book' object has no attribute 'author'

In [76]: one = Book.objects.get(id=1)

In [77]: two = Book.objects.get(id=2)

In [78]: three = Book.objects.get(id=3)

In [79]: third = Author.objects.get(id=3)

In [80]: one.author = third

In [81]: two.author = third

In [82]: three.author = third

In [83]: one.author
Out[83]: <Author: John  >

In [84]: one.save()

In [85]: two.save()

In [86]: three.save()

In [87]: four = Book.objects.get(id=4)

In [88]: four.author = third

In [89]: four.save()

In [90]: four.author
Out[90]: <Author: John  >

In [91]: four.author = speros

In [92]: four.author
Out[92]: <Author: Speros  >

In [93]: four.author.add(third)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-93-4f097508e25f> in <module>()
----> 1 four.author.add(third)

AttributeError: 'Author' object has no attribute 'add'

In [94]: Book.objects.all().authors
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-94-50d2d185a92e> in <module>()
----> 1 Book.objects.all().authors

AttributeError: 'QuerySet' object has no attribute 'authors'

In [95]: four.authors
Out[95]: <django.db.models.fields.related_descriptors.ManyRelatedManager at 0x111ec4b50>

In [96]: mike = Author.objects.get(id=1)

In [97]: book1 = Book.objects.get(id=1)

In [98]: book2 = Book.objects.get(id=2)

In [99]: book1.authors.add(mike)

In [100]: book1.authors.add(speros)

In [101]: book1.authors
Out[101]: <django.db.models.fields.related_descriptors.ManyRelatedManager at 0x111e92110>

In [102]: book1.save()

In [103]: book1.authors
Out[103]: <django.db.models.fields.related_descriptors.ManyRelatedManager at 0x111f26e50>

In [104]: book1authors = book1.authors

In [105]: book1authors
Out[105]: <django.db.models.fields.related_descriptors.ManyRelatedManager at 0x111e94310>

In [106]: for i in book1authors:
     ...:     print i
     ...:     
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-106-dd8accad4833> in <module>()
----> 1 for i in book1authors:
      2     print i
      3 

TypeError: 'ManyRelatedManager' object is not iterable

In [107]: book1.authors.all()
     ...: 
     ...:     
Out[107]: <QuerySet [<Author: Mike Q mike@cd.com>, <Author: Speros  >]>

In [108]: book1authors = book1.authors.all()

In [109]: for i in book1authors:
     ...:     print i
     ...:     
Author object
Author object

In [110]: book1
Out[110]: <Book: C Sharp hi>

In [111]: book2
Out[111]: <Book: Java hi>

In [112]: book3
Out[112]: <Book: Python hi>

In [113]: book4
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-113-a5f14fd5fb26> in <module>()
----> 1 book4

NameError: name 'book4' is not defined

In [114]: book4 = Book.objects.get(id=4)

In [115]: book4
Out[115]: <Book: PHP hi>

In [116]: book5 = Book.objects.get(id=5)

In [117]: book5
Out[117]: <Book: Ruby hi>

In [118]: jadee = Author.objects.get(id=4)

In [119]: jadee
Out[119]: <Author: Jadee  >

In [120]: book1.authors.add(jadee)

In [121]: book2.authors.add(jadee)

In [122]: book3.authors.add(jadee)

In [123]: book4.authors.add(jadee)

In [124]: book5.authors.add(jadee)

In [125]: book5.authors.all()
Out[125]: <QuerySet [<Author: Jadee  >]>

In [126]: book3.authors.add(speros)

In [127]: book3.authors.add(mark)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-127-4773e2edf7f5> in <module>()
----> 1 book3.authors.add(mark)

NameError: name 'mark' is not defined

In [128]: book3.authors.add(mike)

In [129]: book3.authors.all()
Out[129]: <QuerySet [<Author: Jadee  >, <Author: Speros  >, <Author: Mike Q mike@cd.com>]>

In [130]: book3.authors.remove(Jadee)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-130-bccb4cf39637> in <module>()
----> 1 book3.authors.remove(Jadee)

NameError: name 'Jadee' is not defined

In [131]: book3.authors.remove(jadee)

In [132]: book3.authors.all()
Out[132]: <QuerySet [<Author: Speros  >, <Author: Mike Q mike@cd.com>]>

In [133]: book3.authors.add(Author.objects.get(id=5))

In [134]: book3.authors.all()
Out[134]: <QuerySet [<Author: Speros  >, <Author: Mike Q mike@cd.com>, <Author: Jay  >]>

In [135]: k = Author.objects.get(id=5)

In [136]: k
Out[136]: <Author: Jay  >

In [137]: k = "Ketul"

In [138]: k.save()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-138-38f4f919050f> in <module>()
----> 1 k.save()

AttributeError: 'str' object has no attribute 'save'

In [139]: k
Out[139]: 'Ketul'

In [140]: book3.authors.all()
Out[140]: <QuerySet [<Author: Speros  >, <Author: Mike Q mike@cd.com>, <Author: Jay  >]>

In [141]: Author.objects.get(id=5)
Out[141]: <Author: Jay  >

In [142]: k
Out[142]: 'Ketul'

In [143]: k = Author.objects.get(id=5)

In [144]: k
Out[144]: <Author: Jay  >

In [145]: k.first_name = "Ketul"

In [146]: k.save()

In [147]: book3.authors.all()
Out[147]: <QuerySet [<Author: Speros  >, <Author: Mike Q mike@cd.com>, <Author: Ketul  >]>

In [148]: k.books.all()
Out[148]: <QuerySet [<Book: Python hi>]>

In [149]: speros.books.all()
Out[149]: <QuerySet [<Book: C Sharp hi>, <Book: Python hi>]>

In [150]: mike.books.all()
Out[150]: <QuerySet [<Book: C Sharp hi>, <Book: Python hi>]>

In [151]: book2.authors.add(speros)

In [152]: speros.books.all()
Out[152]: <QuerySet [<Book: C Sharp hi>, <Book: Java hi>, <Book: Python hi>]>

In [153]: 
